src/containers/Callers/HistoryPanel.js

```diff
@@ -1,77 +1,76 @@
-            timestamp: this.props.caller.created,
-            timestampStr: this.props.caller.created.toDateString(),
-            type: "Sign Up"
-        });
-        
-        const events = concatenated.sort((a, b) => {
-            return b.timestamp - a.timestamp
-        }).map((el, idx)=> {
-            const color = el.type === "Call" ? "green" : "blue";
-            return  <Timeline.Item color={color} key={idx}>{el.type} on {el.timestampStr}</Timeline.Item>
-        });
-        return (
-            <>
-                <Typography.Title level={3}>Notifications and Calls</Typography.Title>
-                <Typography.Text strong>Notification day of month: {this.props.caller.reminderDayOfMonth}</Typography.Text><br />
-                <Typography.Text>NOTE: This timeline may be incomplete for some callers because notification history only goes back through August, 2019. We did not store notification history prior to then.</Typography.Text>
-                <Divider />
-                <Timeline>{events}</Timeline>
-                <Popconfirm
-                    title={
-                        <Card title="Are you sure?" bordered={false}>
-                            <p>Callers already get automatic call-in notifications once per month.</p>
-                            <p>Use this button for troubleshooting when a caller missed their automatic notification, etc.</p>
-                        </Card>
-                    }
-                    placement="top"
-                    icon={<></>}
-                    onConfirm={(e)=>{this.sendNotification(this.props.caller.callerId)}}
-                    okText="Send"
-                    cancelText="Cancel"
-                >
-                    <Button>Send Notification</Button>
-                </Popconfirm>
-            </>
+const HistoryPanel = ({
+    history,
+    caller,
+}) => { 
+    const events = history.map((item, idx)=> {
+        // TODO: Possibly use an icon here, or expand on the colors
+        let color
+
+        if (item.type === HistoryType.CALL) { 
+            color = 'green' 
+        } else if (item.type === HistoryType.NOTIFICATION) { 
+            color = 'gray'
+        } else if (item.type === HistoryType.SIGN_UP) {
+            color = 'blue'
+        }
+
+        return  (
+            <Timeline.Item key={idx} color={color}>
+                {item.type} on {item.timestampDisplay}
+            </Timeline.Item>
         )
-    }
+    });
 
-    sendNotification = (callerId) => {
-        const requestOptions = {
-          url: `/reminders/${callerId}`,
-          method: 'PUT',
-          headers: { ...authHeader(), 'Content-Type': 'application/json' }
-        };
-        axios(requestOptions).then((response)=>{
-          message.success("Notification sent successfully.")
-        }).catch((e) => {
-            message.error(`Notification failed to send: ${e.message}`);
-        })
-    }
+    return (
```

Comment:
render method is so much more readable with the display logic pulled out. Thank you.

Reply:
Thanks! I've been following the school of thought that your components should generally be functional unless they really need some of the method hooks that class components supply. Since the component only had a render method and the auxiliary `sendNotification`, it seemed like a good candidate for conversion.